README.md

```diff
@@ -20,3 +20,32 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+
+# My Runtime Analysis
+
+In the base case, where the length of the array is 0 or 1, the runtime is constant,
+since in these scenarios, the array is already sorted.
+
+However, when the array size is greater than 1, the permutate function is called. This function generates every possible permutation of the array. With an array that a length of $n$, essentially the algorith will walk through array and swap elements to generate a unique permutation. 
+
+We start with the first permutation, so there is $n$ choices. For the second permutation, there is $n-1$ choice, and so on for the third ($n-2$ choices), fourth ($n-3$ choices), fifth ($n-4$ choices), etc... And since every permutation depends on the one before it, we can write this out as:
+
+$$n*(n-1)*(n-2)*(n-3)... = n!$$
+
+In the worst case scenario, after the permutate function finishes generating all permutations, all the permutations are checked to find the sorted permutation (for the wost case scenario, the sorted permutation is last one to be checked), taking a time complexity of $n$. In the isSolve() method, I have to convert from a string to an array which increases the time complexity to $n^2$ for finding which permutation is sorted. Combining these terms, we get $\Theta(n^2 + n!)$ as worst case complexity for my algorithm. 
```

Comment:
Not following you here -- you can generate all permutations in n time, but `isSolve()` takes quadratic time? I don't see any nested loops in there.